src/scripts/utility-commands/time-and-date.js

```diff
@@ -41,12 +25,18 @@ function getDateTimeFromMessage(text) {
     if (isNaN(jsDate.getTime())) {
         return null;
     }
-    return DateTime.fromISO(jsDate.toISOString());
+    return jsDate;
 }
 
-function getTimeZonesFromMessage(text) {
-    const timezoneRegex = /([A-Za-z]+[\/][A-Za-z0-9_\-+]*)/g;
-    return text.match(timezoneRegex);
+function getFormat(text) {
+    const keys = Object.keys(dynamicDateFormats);
+    let format = null;
+    keys.forEach((possibleFormat) => {
+        if (text.toLowerCase().includes(possibleFormat)) {
+            format = possibleFormat;
+        }
+    });
```

Comment:
Could probably avoid `let` here, plus if there is many `possibleFormat` it only takes last ?